qutebrowser/misc/backendproblem.py

```diff
@@ -414,6 +415,21 @@ def _check_software_rendering(self) -> None:
     def _assert_backend(self, backend: usertypes.Backend) -> None:
         assert objects.backend == backend, objects.backend
 
+    def _reenable_pyinstaller_macos_sandboxing(self) -> None:
+        """WORKAROUND for PyInstaller on macOS disabling sandboxing.
+
+        https://github.com/pyinstaller/pyinstaller/pull/6903
+        https://github.com/qutebrowser/qutebrowser/issues/7278
+        """
+        versions = version.qtwebengine_versions(avoid_init=True)
+        if not hasattr(sys, 'frozen') or \
+           not platform.system() == 'Darwin' or \
+           versions.webengine < utils.VersionNumber(6,):
+            return
+
+        if 'QTWEBENGINE_DISABLE_SANDBOX' in os.environ:
```

Comment:
As [mentioned earlier](https://github.com/qutebrowser/qutebrowser/pull/7500#discussion_r1034470829), I'd actually do this unconditionally:

> I don't think we can easily write a test for it (tests don't run with PyInstaller) - but if the three conditions above are true, we could actually do this without checking `os.environ` first. Then if it's not needed anymore after a PyInstaller upgrade, the `del` will raise a `KeyError`, and that will be caught by the smoke tests before the release (if not before).

Or, perhaps, to be a bit safer, log a warning in an `else:` instead? That at least wouldn't crash if somehow it wasn't set anymore, but should still be enough for us to notice this condition in smoke tests.